src/core/thread/meshcop_dataset_manager.cpp

```diff
@@ -121,6 +121,7 @@ ThreadError DatasetManager::Set(const Timestamp &aTimestamp, const Message &aMes
     SuccessOrExit(error = mNetwork.Set(aMessage, aOffset, aLength));
     mNetwork.SetTimestamp(aTimestamp);
     HandleNetworkUpdate(aFlags);
+    mNetwork.Store();
```

Comment:
I think we should be storing updates to the `mLocal` copy, rather than the `mNetwork` copy.  The `mLocal` copy is the newest dataset that the Thread device is trying to provide the leader.  The `mNetwork` copy is the dataset that the network has currently (which may be older than the `mLocal` copy).